openedx/adg/lms/certificates_override/tests/test_helpers.py

```diff
@@ -0,0 +1,107 @@
+    """
+    Test that `get_arabic_certificate_extra_context` return expected extra context for existing certificates.
+    """
+    course = CourseOverviewFactory(language=ARABIC_LANGUAGE_CODE)
+
+    mocker.patch('openedx.adg.lms.certificates_override.helpers.CourseKey.from_string', return_value=course.id)
+    mocker.patch('openedx.adg.lms.certificates_override.helpers.get_course_by_id', return_value=course)
+
+    certificate = GeneratedCertificateFactory(
+        user=user_with_profile,
+        course_id=course.id,
+        status=CertificateStatuses.downloadable
+    )
+
+    kwargs = {'certificate_uuid': certificate.verify_uuid}
+    request = RequestFactory().get(reverse('certificates:render_cert_by_uuid', kwargs=kwargs))
+
+    mock_resolver_match = mocker.patch.object(request, 'resolver_match')
+    mock_resolver_match.kwargs = kwargs
+
+    actual_gender, actual_date_issued = get_arabic_certificate_extra_context(request, course.id)
+
+    expected_date_format = u'{day} {month} {year}'
+    today = now()
+
+    with translation.override(ARABIC_LANGUAGE_CODE):
+        expected_date = _(expected_date_format).format(  # pylint: disable=translation-of-non-string
+            month=strftime_localized(today, '%B'),
+            day=today.day,
+            year=today.year
+        )
+
+    assert actual_date_issued == expected_date
+    assert actual_gender == user_with_profile.profile.gender
+
+
+@pytest.mark.django_db
+def test_get_arabic_certificate_extra_context_preview_certificate(user_with_profile, mocker, super_user):
+    """
+    Test that `get_arabic_certificate_extra_context` return expected extra context for preview certificate.
+    """
+    course = CourseOverviewFactory(language=ARABIC_LANGUAGE_CODE)
+
+    mocker.patch('openedx.adg.lms.certificates_override.helpers.CourseKey.from_string', return_value=course.id)
+    mocker.patch('openedx.adg.lms.certificates_override.helpers.get_course_by_id', return_value=course)
+
+    kwargs = {'course_id': course.id}
+    preview_url = reverse('certificates:preview_cert', kwargs={'course_id': course.id})
+    preview_url = f'{preview_url}?preview=honor'
+
+    request = RequestFactory().get(preview_url)
+    request.user = super_user
+
+    mock_resolver_match = mocker.patch.object(request, 'resolver_match')
+    mock_resolver_match.kwargs = kwargs
+
+    actual_gender, actual_date_issued = get_arabic_certificate_extra_context(request, course.id)
+
+    expected_date_format = u'{day} {month} {year}'
+    today = now()
+
+    with translation.override(ARABIC_LANGUAGE_CODE):
+        expected_date = _(expected_date_format).format(  # pylint: disable=translation-of-non-string
+            month=strftime_localized(today, '%B'),
+            day=today.day,
+            year=today.year
+        )
+
+    assert actual_date_issued == expected_date
+    assert actual_gender == user_with_profile.profile.gender
```

Comment:
We can combine the two test cases as most of the code is being repeated on the basis of user type.
```
if super_user condition:
    kwargs = {'course_id': course.id}
    preview_url = reverse('certificates:preview_cert', kwargs={'course_id': course.id})
    preview_url = f'{preview_url}?preview=honor'

    request = RequestFactory().get(preview_url)
    request.user = super_user
else:
    certificate = GeneratedCertificateFactory(
        user=user_with_profile,
        course_id=course.id,
        status=CertificateStatuses.downloadable
    )

    kwargs = {'certificate_uuid': certificate.verify_uuid}
    request = RequestFactory().get(reverse('certificates:render_cert_by_uuid', kwargs=kwargs))

Reply:
@javeriaejaz-arbisoft as I replied to Anas, The similarities are mocking and assertion. The core code is testing different views. If you still think that we  should merge these then I can. Actually initially I created 1 test and used parametrization but then splitted that test intentionally.